src/cc_catalog_airflow/dags/provider_api_scripts/jamendo.py

```diff
@@ -0,0 +1,322 @@
+"""
+Content Provider:       Jamendo
+
+ETL Process:            Use the API to identify all CC-licensed images.
+
+Output:                 TSV file containing the image, the respective
+                        meta-data.
+
+Notes:                  https://api.jamendo.com/v3.0/tracks/
+                        No rate limit specified.
+"""
+import os
+import logging
+from urllib.parse import urlparse
+
+from common import DelayedRequester, AudioStore
+from common.licenses.licenses import get_license_info
+from common.urls import rewrite_redirected_url
+from util.loader import provider_details as prov
+
+# On Jamendo Music, you can enjoy a wide catalog of more than 500,000 tracks
+# shared by 40,000 artists from over 150 countries all over the world.
+
+# Audio quality: uploaded as WAV/ FLAC/ AIFF
+# bit depth: 16/24
+# sample rate: 44.1 or 48 kHz
+# channels: 1/2
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+LIMIT = 200  # number of items per page in API response
+DELAY = 1  # in seconds
+RETRIES = 3
+
+HOST = 'jamendo.com'
+ENDPOINT = f'https://api.{HOST}/v3.0/tracks'
+PROVIDER = prov.JAMENDO_DEFAULT_PROVIDER
+APP_KEY = os.getenv('JAMENDO_APP_KEY', 'dd3c1077')
+
+HEADERS = {
+    "Accept": "application/json",
+}
+DEFAULT_QUERY_PARAMS = {
+    # jsonpretty can have invalid characters in json:
+    # \u0009 tab breaks the json
+    'format': 'json',
+    'client_id': APP_KEY,
+    'include': 'musicinfo licenses stats lyrics',
+    'imagesize': 200,
+    'limit': 200,
+    'audioformat': 'mp32',
+}
+
+delayed_requester = DelayedRequester(DELAY)
+audio_store = AudioStore(provider=PROVIDER)
+
+
+def main():
+    """
+    This script pulls the data for a given date from the Jamendo,
+    and writes it into a .TSV file to be eventually read
+    into our DB.
+    """
+
+    logger.info("Begin: Jamendo script")
+    image_count = _get_items()
+    logger.info(f"Total images pulled: {image_count}")
```

Comment:
```suggestion
    logger.info("Begin: Jamendo script")
    image_count = _get_items()
    audio_store.commit()
    logger.info(f"Total images pulled: {image_count}")
```
Shouldn't it always commit in the end? If the last page/batch doesn't reach 100 items it may be lost. This happened to me with StockSnap, I didn't let it get 100 items so it did not write the file.

Reply:
I haven't realized this before! I will add `commit` to the template script, too.